common/djangoapps/student/views.py

```diff
@@ -1564,6 +1564,32 @@ def login_user(request, error=""):  # pylint: disable=too-many-statements,unused
         if user_found_by_email_lookup and LoginFailures.is_feature_enabled():
             LoginFailures.increment_lockout_counter(user_found_by_email_lookup)
 
+        # TODO: Remove Django 1.11 upgrade shim
+        # SHIM: In Django 1.10+ the user will not authenticate if it is not active (after the initial account creation,
+        # see NEW_USER_AUTH_BACKEND for how we handle that). This code is duplicated below for the < 1.10 case where
+        # the user is successfully authenticated, but not active. To remove this shim, remove the version check here
+        # and the duplicated code at the end of this function.
+        #
+        # Note that this will be triggered for ALL inactive users, regardless of whether the password is correct. In
+        # either case they need to activate before continuing so password status is moot.
```

Comment:
My brain is starting to hurt trying to trace through all this.  The `login_user` method is now 251 lines long; can the duplicated portion be broken out into a separate function, that's just called in 2 different places?  Feel free to break out other blocks while you're at it...